frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx

```diff
@@ -475,24 +512,24 @@ export function AnsibleAutomationsForm(props: {
             <AcmDataFormPage formData={formData} mode={isViewing ? 'details' : isEditing ? 'form' : 'wizard'} />
             <EditAnsibleJobModal
                 ansibleJob={editAnsibleJob}
+                ansibleSelection={ansibleSelection}
                 setAnsibleJob={updateAnsibleJob}
-                ansibleJobList={editAnsibleJobList?.jobs}
+                ansibleCredentials={ansibleCredentials}
+                ansibleTowerTemplateList={AnsibleTowerJobTemplateList}
             />
         </Fragment>
     )
 }
 
 function EditAnsibleJobModal(props: {
+    ansibleSelection?: string
+    ansibleCredentials: ProviderConnection[]
+    ansibleTowerTemplateList: string[] | undefined
     ansibleJob?: ClusterCuratorAnsibleJob
     setAnsibleJob: (ansibleJob?: ClusterCuratorAnsibleJob, old?: ClusterCuratorAnsibleJob) => void
-    ansibleJobList?: ClusterCuratorAnsibleJob[]
 }) {
     const { t } = useTranslation(['common', 'cluster'])
     const [ansibleJob, setAnsibleJob] = useState<ClusterCuratorAnsibleJob | undefined>()
-    let ansibleJobList: string[]
-    if (props.ansibleJobList)
-        ansibleJobList = props.ansibleJobList.filter((job) => ansibleJob !== job).map((ansibleJob) => ansibleJob.name)
-    useEffect(() => setAnsibleJob(props.ansibleJob), [props.ansibleJob])
```

Comment:
Sorry @chenz4027, I went a touch too far with this one.
This line `useEffect(() => setAnsibleJob(props.ansibleJob), [props.ansibleJob])`
actually needs to be here! It's what's currently braking the build.

Reply:
Ahh I see thanks for the catch!